layers/ip6.go

```diff
@@ -563,29 +608,85 @@ type IPv6Routing struct {
 // LayerType returns LayerTypeIPv6Routing.
 func (i *IPv6Routing) LayerType() gopacket.LayerType { return LayerTypeIPv6Routing }
 
-func decodeIPv6Routing(data []byte, p gopacket.PacketBuilder) error {
-	base, err := decodeIPv6ExtensionBase(data, p)
+// SerializeTo implementation according to gopacket.SerializableLayer
+func (i *IPv6Routing) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
+	bytes, err := b.PrependBytes(8 + len(i.SourceRoutingIPs)*16)
 	if err != nil {
 		return err
 	}
-	i := &IPv6Routing{
-		ipv6ExtensionBase: base,
-		RoutingType:       data[2],
-		SegmentsLeft:      data[3],
-		Reserved:          data[4:8],
+	if opts.FixLengths {
+		i.HeaderLength = uint8(len(i.SourceRoutingIPs) * 2)
 	}
+	bytes[0] = byte(i.NextHeader)
+	bytes[1] = i.HeaderLength
+	bytes[2] = i.RoutingType
+	bytes[3] = i.SegmentsLeft
 	switch i.RoutingType {
-	case 0: // Source routing
-		if (i.ActualLength-8)%16 != 0 {
-			return fmt.Errorf("Invalid IPv6 source routing, length of type 0 packet %d", i.ActualLength)
+	case 0:
+		copy(bytes[4:], i.Reserved)
+	case 4:
+		if opts.FixLengths {
+			i.LastEntry = uint8(len(i.SourceRoutingIPs))
```

Comment:
```suggestion
			i.LastEntry = uint8(len(i.SourceRoutingIPs)) - 1
```
According to RFC8754: "Last Entry:  contains the index (zero based), in the Segment List, of the last element of the Segment List."